### PR TITLE
fix(security): potential arbitary code execution vulnerability

### PR DESCRIPTION
**Pull request by quantum9Innovation**
## **User description**
sanitize all user input in zenodo uploader

fixes a medium-severity security vulnerability first identified by Snyk

___

## **Type**
bug_fix, enhancement, configuration changes

___

## **Description**
- Added `shellescape` to sanitize user input in the Zenodo uploader script, addressing a potential arbitrary code execution vulnerability.
- Enhanced the shell script with a shebang and the TypeScript compilation flag `esModuleInterop`.
- Configured CodeClimate analysis with various plugins and specified exclude patterns.
- Updated dependencies to include `@types/shell-escape` and `shell-escape` for input sanitization.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>zenodo.ts</strong><dd><code>Sanitize User Input in Zenodo Uploader Script</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
scripts/zenodo.ts

<li>Added shellescape to sanitize user input before running commands to <br>prevent arbitrary code execution.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Team-1280/eeXiv/pull/36/files#diff-b09b8d8b185d340416d75c3c3c4954700a2cdd11807ca5bef8d55f41973536b7">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>run.sh</strong><dd><code>Enhance Shell Script with Shebang and TypeScript Compilation Flag</code></dd></summary>
<hr>
      
scripts/run.sh

<li>Added shebang to specify the script interpreter.<br> <li> Enabled esModuleInterop flag in TypeScript compilation command.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Team-1280/eeXiv/pull/36/files#diff-bb563bfe483a3a9b1649cf50d2246475b9565947f5226b94fa0d72c9786ab298">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Configuration changes
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>.codeclimate.yml</strong><dd><code>Configure CodeClimate Analysis</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
.codeclimate.yml

<li>Configured various CodeClimate plugins including eslint, shellcheck, <br>and markdownlint.<br> <li> Specified exclude patterns for CodeClimate analysis.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Team-1280/eeXiv/pull/36/files#diff-e97904d8a0443f46ad2c8ceb5fbe5355e52f6be540378ab8ea1e98870395d7de">+31/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Dependencies
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>package-lock.json</strong><dd><code>Update Dependencies for Shell Command Sanitization</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
package-lock.json

- Added `@types/shell-escape` and `shell-escape` packages.

</details>
    

  </td>
  <td><a href="https://github.com/Team-1280/eeXiv/pull/36/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519">+14/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>package.json</strong><dd><code>Add Dependencies for User Input Sanitization</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
package.json

- Added `@types/shell-escape` and `shell-escape` to dependencies.

</details>
    

  </td>
  <td><a href="https://github.com/Team-1280/eeXiv/pull/36/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: scripts/run.sh [hunk 1]
```diff
@@ -1 +1,2 @@
-npx tsc zenodo.ts && node zenodo.js $1 $2
+#!/usr/bin/bash
+npx tsc --esModuleInterop zenodo.ts && node zenodo.js $1 $2
```
[the remark this answers is not in the document]
Comment (quantum9Innovation) [reply scripts/run.sh:2]: Bash brainrot; will not implement

File: scripts/run.sh [hunk 2]
```diff
@@ -2,0 +1,2 @@
+#!/usr/bin/bash
```
[the remark this answers is not in the document]
Comment (quantum9Innovation) [reply scripts/run.sh:1]: Won't fix cope L ratio

**Review by quantum9Innovation — COMMENTED** [review]
@couscousdude review this critical pending security update before it is merged later today

**Review by couscousdude — APPROVED** [review]
LGTM 🚢

**Pull request merged**